pkg/controllers/state/cluster.go

```diff
@@ -90,6 +104,14 @@ func (c *Cluster) Synced(ctx context.Context) (synced bool) {
 	// Set the metric to whatever the result of the Synced() call is
 	defer func() {
 		clusterStateSynced.Set(lo.Ternary[float64](synced, 1, 0))
+		if synced {
+			c.lastSynced = time.Now()
+		}
+		// If we have been out of sync for a while, and we haven't logged about it recently, log it
+		if time.Since(c.lastSynced) > AbnormalClusterSyncDuration && c.cm.HasChanged("ClusterSynced", synced) {
```

Comment:
musing: I wonder if we could encapsulate this behavior to reduce the complexity of our Synced() cluster function.

```go
// ErrorStateMonitor monitors how long something maintains a "target" value. If that target value is maintained for long enough (`errorThreshold`), it fires an error to represent that the component hasn't recovered from the error state and will fire this error at every `interval`.
type ErrorStateMonitor struct {
  target interface{}
  errorThreshold time.Duration
  interval time.Duration
}
```